include/deal.II/lac/petsc_snes.h

```diff
@@ -233,13 +233,27 @@ namespace PETScWrappers
    * model reductions using the energy functional. In such cases, users can
    * set an optional NonlinearSolver::energy callback.
    *
+   * @dealiiConceptRequires{(concepts::is_dealii_petsc_vector_type<VectorType>
+   * || std::constructible_from<VectorType,Vec>) &&
+   *   (concepts::is_dealii_petsc_matrix_type<PMatrixType> ||
+   *    std::constructible_from<PMatrixType,Mat>) &&
+   *   (concepts::is_dealii_petsc_matrix_type<AMatrixType> ||
+   *    std::constructible_from<AMatrixType, Mat>)}
+   *
    * @ingroup PETScWrappers
    */
   template <typename VectorType  = PETScWrappers::VectorBase,
             typename PMatrixType = PETScWrappers::MatrixBase,
             typename AMatrixType = PMatrixType>
-  DEAL_II_CXX20_REQUIRES((concepts::is_dealii_petsc_vector_type<VectorType> ||
-                          std::constructible_from<VectorType, Vec>))
+  DEAL_II_CXX20_REQUIRES(
+    (concepts::is_dealii_petsc_vector_type<VectorType> ||
+     std::constructible_from<
+       VectorType,
+       Vec>)&&(concepts::is_dealii_petsc_matrix_type<PMatrixType> ||
+               std::constructible_from<
+                 PMatrixType,
+                 Mat>)&&(concepts::is_dealii_petsc_matrix_type<AMatrixType> ||
+                         std::constructible_from<AMatrixType, Mat>))
```

Comment:
I'll readily admit that this is not wrapped in a way that makes it easy to read :-(

Reply:
You can always use `clang-format off`.

Reply:
or update the clang-version :-)

Reply:
Not a battle for this PR to fight :-)